app/models/offer.rb

```diff
@@ -119,6 +119,15 @@ def scheduled?
     end
   end
 
+  def self.find_by_ggv_uuid(ggv_uuid)
+    begin
+      offer = GogovanOrder.find_by(ggv_uuid: ggv_uuid).delivery.offer
+    rescue Exception => e
+      raise ActiveRecord::RecordNotFound
+    end
+    with_eager_load.find(offer.id)
+  end
```

Comment:
Sure, I'll move it into the GogovanOrder class.
In order to handle scenario where driver bookmarked the link(or then offer has been cancelled) then we have to show 404 popup to driver, as @mattgow mentioned in the story GCW-230.

Thanks